src/screens/SignTransaction/SignWithColdCard.tsx

```diff
@@ -89,19 +89,34 @@ function SignWithColdCard({ route }: { route }) {
 
   const receiveFromColdCard = async () =>
     withNfcModal(async () => {
-      dispatch(
-        healthCheckStatusUpdate([
-          {
-            signerId: signer.masterFingerprint,
-            status: hcStatusType.HEALTH_CHECK_SIGNING,
-          },
-        ])
-      );
       if (!isMultisig) {
         const { txn } = await receiveTxHexFromColdCard();
         dispatch(updatePSBTEnvelops({ xfp: vaultKey.xfp, txHex: txn }));
       } else {
         const { psbt } = await receivePSBTFromColdCard();
+        if (isRemoteKey) {
+          dispatch(
+            healthCheckStatusUpdate([
+              {
+                signerId: signer.masterFingerprint,
+                status: hcStatusType.HEALTH_CHECK_SIGNING,
+              },
+            ])
+          );
+          navigation.dispatch(
```

Comment:
So it first signs the PSBT and only then shows the confirmation sceen? I think that should be reversed no?

Reply:
@ben-kaufman 
ShowPSBT is for displaying the signed QR of psbt, 
tnx details are shown first then signed and after that its displayed in ShowPSBT screen